stdlib/cache/src/main/ballerina/src/cache/cache.bal

```diff
@@ -14,336 +14,302 @@
-    public function put(string key, any value) {
-        // We need to synchronize this process otherwise concurrency might cause issues.
-         lock {
-            int cacheCapacity = self.capacity;
-            int cacheSize = self.entries.length();
-
-            // If the current cache is full, evict cache.
-            if (cacheCapacity <= cacheSize) {
-                self.evict();
+    # + key - Key of the cached value
+    # + value - Value to be cached
+    # + maxAgeInSeconds - The value in seconds which the cache entry is valid. '-1' means, the entry is valid forever.
+    public function put(string key, any value, int maxAgeInSeconds = -1) {
+        lock {
+            // If the current cache is full (i.e. size = capacity), evict cache.
+            if (self.size() == self.capacity) {
+                evict(self.entries, self.list, self.evictionPolicy, self.capacity, self.evictionFactor);
             }
-            // Add the new cache entry.
-            int time = time:currentTime().time;
-            CacheEntry entry = { value: value, lastAccessedTime: time };
-            self.entries[key] = entry;
-
-            // If the UUID is not found, that means that cache was removed after being empty. So we need to create a
-            // new cache with the current cache object.
-            if (!cacheMap.hasKey(self.uuid)) {
-                cacheMap[self.uuid] = self;
+
+            // Calculate the `expTime` of the cache entry based on the `maxAgeInSeconds` property and
+            // `defaultMaxAgeInSeconds` property.
+            int calculatedExpTime = -1;
+            if (maxAgeInSeconds != -1 && maxAgeInSeconds > 0) {
+                calculatedExpTime = time:nanoTime() + (maxAgeInSeconds * 1000 * 1000 * 1000);
+            } else {
+                if (self.defaultMaxAgeInSeconds != -1) {
+                    calculatedExpTime = time:nanoTime() + (self.defaultMaxAgeInSeconds * 1000 * 1000 * 1000);
+                }
             }
-         }
-    }
 
-    # Evicts the cache when the cache is full.
-    function evict() {
-        int maxCapacity = self.capacity;
-        float ef = self.evictionFactor;
-        int numberOfKeysToEvict = <int>(maxCapacity * ef);
-        // Get the above number of least recently used cache entry keys from the cache
-        string[] cacheKeys = self.getLRUCacheKeys(numberOfKeysToEvict);
-        // Iterate through the map and remove entries.
-        foreach var c in cacheKeys {
-            // These cache values are ignored. So it is not needed to check the return value for the remove function.
-            var tempVar = self.entries.remove(c);
+            CacheEntry entry = {
+                key: key,
+                data: value,
+                expTime: calculatedExpTime
+            };
+
+            if (self.hasKey(key)) {
+                Node node = self.entries.get(key);
+                putOnEvictionPolicy(self.evictionPolicy, self.list, node, true);
```

Comment:
Here we need to remove the old node from the cache and add the new node. seems like we are adding the same old node into the cache

Reply:
Yes. Fixed it with https://github.com/ballerina-platform/ballerina-lang/pull/21308/commits/4cf23f89727e3e6593d6375baa00cf0e9b75329b